### PR TITLE
Parallelize GTTA API

### PR DESCRIPTION
**Pull request by marktwtn**
# Description

The `synchronized` keyword has been removed for improving the response time and throughput of the GTTA requests.
The pull request is based on the experiment result of issue #973.

IRI 1.5.3 - with synchronization
gtta parallel test on testnet - 1 5 3

Maximum response time: 23731 ms
Throughput: 2,373.699 / minute

IRI 1.5.5 - without synchronization
gtta parallel test on testnet - no sync

Maximum response time: 3694 ms
Throughput: 15,011.258 / minute

## Type of change

- Performance

# How Has This Been Tested?

- Use `jMeter` to send a 1000 GTTA calls with the same depth(15) at the same time and log the time it takes to response to all of them.

# Checklist:

_Please delete items that are not relevant._
- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by jserv** [comment]
Please revise the Git commit messages. Reference: https://chris.beams.io/posts/git-commit/
At least motivations should be addressed.

**Comment by jserv** [comment]
@jakubcech, can you review this PR?

**Comment by GalRogozinski** [comment]
The reason we are blocking multiple calls to GTTA is to prevent DOS on the node.
During the random walks entails sig validation which is a heavy operation.

According to #973 no DOS happened since @marktwtn claimed that the latest solid milestone updated correctly. Still IF needs to validate that as a proper regression test that will always run on the network.

Before we merge I want a nightly test that does:
We need to set up on mainnet or spamnet (@geminoz this could be a great test to do on spamnet) a node without `synchronized` and a control node with `synchronized`. We need to setup GTTA spammers (preferrably JMeter which will report us stats) which will work on both nodes. Lastly, we ascertain that neither nodes start to fall out of sync.

In other words I want #973 implemented as a nightly test as a prerequisite to merge.

**Comment by DyrellC** [comment]
I can confirm that parallel calls for GTTA with a high enough volume of calls can cause OOM errors fairly quickly on the node. When they are sent in batches it's not too bad, but in the event you send 1000 at once and your node has a low allocation of memory, it's likely you'll run into those OOM errors. 

image

image
